app/controllers/checkout_controller.rb

```diff
@@ -0,0 +1,144 @@
+class CheckoutController < ApplicationController
```

Comment:
Too much logic in the controller. Please refactor it. You can create your own service objects or you a gem like: rectify, interactor, trailblazer etc.
http://rubygarage.github.io/slides/refactoring-rails#/